### PR TITLE
pythonPackages.azure-keyvault: init at 1.1.0

### PR DESCRIPTION
**Pull request by peterromfeldhk**
pythonPackages.msrestazure: init at 0.6.0
pythonPackages.msrest: init at 0.6.4

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/azure-keyvault/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    extension = "zip";
+    sha256 = "0jfxm8lx8dzs3v2b04ljizk8gfckbm5l2v86rm7k0npbfvryba1p";
+  };
+
+  propagatedBuildInputs = [ azure-common msrest msrestazure ];
+
+  doCheck = false;
```
Comment (dotlambda) [inline pkgs/development/python-modules/azure-keyvault/default.nix]: Why?

File: pkgs/development/python-modules/azure-keyvault/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  meta = with lib; {
+    description = "This is the Microsoft Azure Key Vault Client Library";
+    license = with licenses; [ mit ];
+  };
```
Comment (dotlambda) [inline pkgs/development/python-modules/azure-keyvault/default.nix]: Does *not* fit CONTRIBUTING.md :(

File: pkgs/development/python-modules/azure-keyvault/default.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    extension = "zip";
+    sha256 = "0jfxm8lx8dzs3v2b04ljizk8gfckbm5l2v86rm7k0npbfvryba1p";
+  };
+
+  propagatedBuildInputs = [ azure-common msrest msrestazure ];
+
+  doCheck = false;
+
+  meta = with lib; {
+    description = "This is the Microsoft Azure Key Vault Client Library";
```
Comment (dotlambda) [inline pkgs/development/python-modules/azure-keyvault/default.nix]:
```suggestion
    description = "Microsoft Azure Key Vault Client Library";
```

File: pkgs/development/python-modules/msrest/default.nix [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    inherit pname version;
+    sha256 = "0rnj79fw571hqwj5z8zsraw3aq9lbh0rbc1zykcjrmcqxi5xbbax";
+  };
+
+  propagatedBuildInputs = [ isodate requests requests_oauthlib ];
+
+  doCheck = false;
+
+  meta = with lib; {
+    description = "MS: AutoRest: Python Client Runtime";
```
Comment (dotlambda) [inline pkgs/development/python-modules/msrest/default.nix]: please be more descriptive

File: pkgs/development/python-modules/msrest/default.nix [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  disabled = isPyPy;
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0rnj79fw571hqwj5z8zsraw3aq9lbh0rbc1zykcjrmcqxi5xbbax";
+  };
+
+  propagatedBuildInputs = [ isodate requests requests_oauthlib ];
+
+  doCheck = false;
```
Comment (dotlambda) [inline pkgs/development/python-modules/msrest/default.nix]: Why?

File: pkgs/development/python-modules/msrestazure/default.nix [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  disabled = isPyPy;
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "06s04f6nng4na2663kc12a3skiaqb631nscjfwpsrx4lzkf8bccr";
+  };
+
+  propagatedBuildInputs = [ adal msrest ];
+
+  doCheck = false;
```
Comment (dotlambda) [inline pkgs/development/python-modules/msrestazure/default.nix]: Why?

File: pkgs/development/python-modules/msrestazure/default.nix [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+  meta = with lib; {
+    description = "MS: AutoRest: Python Client Runtime - Azure Module";
```
Comment (dotlambda) [inline pkgs/development/python-modules/msrestazure/default.nix]: also not optimal

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/msrestazure/default.nix (hunk 7 above)
Comment (peterromfeldhk) [reply pkgs/development/python-modules/msrestazure/default.nix]: this is a dependency of azure-keyvault and has no description.. i mean i can come up with some bs myself but i dont use this crap directly and would love to rip out the rest of MS...

Comment (dotlambda) [reply pkgs/development/python-modules/msrestazure/default.nix]: Please do come up with one yourself.

File: pkgs/development/python-modules/msrest/default.nix (hunk 4 above)
Comment (peterromfeldhk) [reply pkgs/development/python-modules/msrest/default.nix]: maybe resolved

File: pkgs/development/python-modules/msrestazure/default.nix (hunk 7 above)
Comment (peterromfeldhk) [reply pkgs/development/python-modules/msrestazure/default.nix]: maybe resolved

**Comment by peterromfeldhk** [comment]
i am getting
```
collision between `/nix/store/vwxrdazjg2z2v0z2m5njlvysxv3bhg07-python3.7-azure-common-1.1.16/lib/python3.7/site-packages/azure/__init__.py' and `/nix/store/vx993ylcavaphrwz483zp1m01js6wjga-python3.7-azure-keyvault-1.1.0/lib/python3.7/site-packages/azure/__init__.py'
```

not sure how to fix :(

File: pkgs/development/python-modules/azure-keyvault/default.nix [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  propagatedBuildInputs = [ azure-common msrest msrestazure ];
+
+  # no tests in pypi package, not sure how to do fetchFromGitHub with this monorepo
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Microsoft Azure Key Vault Client Library";
+    homepage = https://github.com/Azure/azure-sdk-for-python/tree/master/azure-keyvault;
+    maintainers= with maintainers; [ peterromfeldhk ];
+    license = with licenses; [ mit ];
```
[the remark this answers is not in the document]
Comment (peterromfeldhk) [reply pkgs/development/python-modules/azure-keyvault/default.nix]: seriously?

i really like your input and it helps me to make better MR's... but this?

[the remark this answers is not in the document]
Comment (dotlambda) [reply pkgs/development/python-modules/azure-keyvault/default.nix]: I wouldn't have requested it if there were no other things to change.

File: pkgs/development/python-modules/msrestazure/default.nix [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+  };
+
+  propagatedBuildInputs = [ adal msrest ];
+
+  # TypeError: async_poller() missing 4 required positional arguments: 'client',
+  # 'initial_response', 'deserialization_callback', and 'polling_method'
+  doCheck = false;
+
+  meta = with lib; {
+    description = "MS wrapper around wrapper around requests without description";
```
[the remark this answers is not in the document]
Comment (peterromfeldhk) [reply pkgs/development/python-modules/msrestazure/default.nix]: yes they are a dependency of one lib i need :(

File: pkgs/development/python-modules/msrest/default.nix [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+    owner = "Azure";
+    repo = "${pname}-for-python";
+    rev = "v${version}";
+    sha256 = "0ilrc06qq0dw4qqzq1dq2vs6nymc39h19w52dwcyawwfalalnjzi";
+  };
+
+  propagatedBuildInputs = [ isodate requests requests_oauthlib ];
+
+  # TypeError: async_poller() missing 4 required positional arguments: 'client',
+  # 'initial_response', 'deserialization_callback', and 'polling_method'
```
[the remark this answers is not in the document]
Comment (peterromfeldhk) [reply pkgs/development/python-modules/msrest/default.nix]: i tried to add aiodns, tried using 3.6 and looking through the docs and travic-ci.yaml but could not figure out whats missing :(
```
Traceback (most recent call last):
  File "nix_run_setup", line 8, in <module>
    exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
  File "setup.py", line 61, in <module>
    'aiodns'
  File "/nix/store/z52q964awi2ix9pivyayx2vkyjhvq6hk-python3.6-bootstrapped-pip-18.1/lib/python3.6/site-packages/setuptools/__init__.py", line 143, in setup
    return distutils.core.setup(**attrs)
  File "/nix/store/9xa9i83pfcbdhnks928h5d24a1ks9d7j-python3-3.6.8/lib/python3.6/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/nix/store/9xa9i83pfcbdhnks928h5d24a1ks9d7j-python3-3.6.8/lib/python3.6/distutils/dist.py", line 955, in run_commands
    self.run_command(cmd)
  File "/nix/store/9xa9i83pfcbdhnks928h5d24a1ks9d7j-python3-3.6.8/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/nix/store/z52q964awi2ix9pivyayx2vkyjhvq6hk-python3.6-bootstrapped-pip-18.1/lib/python3.6/site-packages/setuptools/command/test.py", line 228, in run
    self.run_tests()
  File "/nix/store/z52q964awi2ix9pivyayx2vkyjhvq6hk-python3.6-bootstrapped-pip-18.1/lib/python3.6/site-packages/setuptools/command/test.py", line 250, in run_tests
    exit=False,
  File "/nix/store/9xa9i83pfcbdhnks928h5d24a1ks9d7j-python3-3.6.8/lib/python3.6/unittest/main.py", line 94, in __init__
    self.parseArgs(argv)
  File "/nix/store/9xa9i83pfcbdhnks928h5d24a1ks9d7j-python3-3.6.8/lib/python3.6/unittest/main.py", line 118, in parseArgs
    self._do_discovery(argv[2:])
  File "/nix/store/9xa9i83pfcbdhnks928h5d24a1ks9d7j-python3-3.6.8/lib/python3.6/unittest/main.py", line 229, in _do_discovery
    self.test = loader.discover(self.start, self.pattern, self.top)
  File "/nix/store/9xa9i83pfcbdhnks928h5d24a1ks9d7j-python3-3.6.8/lib/python3.6/unittest/loader.py", line 341, in discover
    tests = list(self._find_tests(start_dir, pattern))
  File "/nix/store/9xa9i83pfcbdhnks928h5d24a1ks9d7j-python3-3.6.8/lib/python3.6/unittest/loader.py", line 398, in _find_tests
    full_path, pattern, namespace)
  File "/nix/store/9xa9i83pfcbdhnks928h5d24a1ks9d7j-python3-3.6.8/lib/python3.6/unittest/loader.py", line 475, in _find_test_path
    tests = self.loadTestsFromModule(package, pattern=pattern)
  File "/nix/store/z52q964awi2ix9pivyayx2vkyjhvq6hk-python3.6-bootstrapped-pip-18.1/lib/python3.6/site-packages/setuptools/command/test.py", line 54, in loadTestsFromModule
    tests.append(self.loadTestsFromName(submodule))
  File "/nix/store/9xa9i83pfcbdhnks928h5d24a1ks9d7j-python3-3.6.8/lib/python3.6/unittest/loader.py", line 190, in loadTestsFromName
    return self.loadTestsFromModule(obj)
  File "/nix/store/z52q964awi2ix9pivyayx2vkyjhvq6hk-python3.6-bootstrapped-pip-18.1/lib/python3.6/site-packages/setuptools/command/test.py", line 54, in loadTestsFromModule
    tests.append(self.loadTestsFromName(submodule))
  File "/nix/store/9xa9i83pfcbdhnks928h5d24a1ks9d7j-python3-3.6.8/lib/python3.6/unittest/loader.py", line 204, in loadTestsFromName
    test = obj()
TypeError: async_poller() missing 4 required positional arguments: 'client', 'initial_response', 'deserialization_callback', and 'polling_method'
```

**Comment by dotlambda** [comment]
```diff
diff --git a/pkgs/development/python-modules/msrest/default.nix b/pkgs/development/python-modules/msrest/default.nix
index 0abf1508113..7e16d7f5859 100644
--- a/pkgs/development/python-modules/msrest/default.nix
+++ b/pkgs/development/python-modules/msrest/default.nix
@@ -1,5 +1,9 @@
 { lib, buildPythonPackage, fetchFromGitHub
-, isodate, requests, requests_oauthlib }:
+, pythonOlder, pythonAtLeast, isPy3k
+, requests, requests_oauthlib, isodate, certifi
+, enum34, typing, aiohttp, aiodns
+, pytest, httpretty, mock
+}:
 
 buildPythonPackage rec {
   pname = "msrest";
@@ -12,14 +16,20 @@ buildPythonPackage rec {
     sha256 = "0ilrc06qq0dw4qqzq1dq2vs6nymc39h19w52dwcyawwfalalnjzi";
   };
 
-  propagatedBuildInputs = [ isodate requests requests_oauthlib ];
+  propagatedBuildInputs = [ requests requests_oauthlib isodate certifi ]
+    ++ lib.optional (pythonOlder "3.4") enum34
+    ++ lib.optional (pythonOlder "3.5") typing
+    ++ lib.optionals (pythonAtLeast "3.5") [ aiohttp aiodns ];
 
-  # TypeError: async_poller() missing 4 required positional arguments: 'client',
-  # 'initial_response', 'deserialization_callback', and 'polling_method'
-  doCheck = false;
+  checkInputs = [ pytest httpretty ]
+    ++ lib.optional (!isPy3k) mock;
+
+  checkPhase = ''
+    pytest tests/
+  '';
 
   meta = with lib; {
-    description = "Microsoft wrapper around requests";
+    description = "AutoRest swagger generator Python client runtime";
     homepage = https://github.com/Azure/msrest-for-python;
     maintainers = with maintainers; [ peterromfeldhk ];
     license = licenses.mit;
```
There are still some errors I'll leave to you but that was not very difficult to do.

Also, I found out you didn't add the dependencies for older Python versions: https://github.com/Azure/msrest-for-python/blob/master/setup.py#L57-L58. Please check this for future and existing PRs.

Their setup.py features a description we can use: https://github.com/Azure/msrest-for-python/blob/master/setup.py#L36.

Finally, please make separate commits as I requested and fix the eval failure detected by ofborg: https://gist.github.com/GrahamcOfBorg/e76145b890e5d2a14ea9b2bfe84c6882

**Comment by mmahut** [comment]
Are there any updates on this pull request, please?

**Comment by yorickvP** [comment]
added in #60435

**Pull request closed without merging**